homeassistant/components/blueprint/importer.py

```diff
@@ -0,0 +1,177 @@
+"""Import logic for blueprint."""
+from dataclasses import dataclass
+import re
+from typing import Optional
+
+import voluptuous as vol
+import yarl
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.util import yaml
+
+from .models import Blueprint
+from .schemas import is_blueprint_config
+
+COMMUNITY_TOPIC_PATTERN = re.compile(
+    r"^https://community.home-assistant.io/t/[a-z-]+/(?P<topic>\d+)(?:/(?P<post>\d+)|)$"
+)
+
+COMMUNITY_CODE_BLOCK = re.compile(
+    r'<code class="lang-(?P<syntax>[a-z]+)">(?P<content>(?:.|\n)*)</code>', re.MULTILINE
+)
+
+GITHUB_FILE_PATTERN = re.compile(
+    r"^https://github.com/(?P<repository>.+)/blob/(?P<path>.+)$"
+)
+GITHUB_RAW_FILE_PATTERN = re.compile(r"^https://raw.githubusercontent.com/")
+
+COMMUNITY_TOPIC_SCHEMA = vol.Schema(
+    {
+        "slug": str,
+        "title": str,
+        "post_stream": {"posts": [{"updated_at": cv.datetime, "cooked": str}]},
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+@dataclass
+class ImportedBlueprint:
```

Comment:
Why not use attr and his nice `slot=True` feature?

Reply:
dataclass is built-in and pretty much the same :)

Reply:
I could add `frozen=True` here. Which is a similar feature.

Reply:
For memory footprint, it would be better if we not need to extend the object at runtime.

Reply:
This object will remain in memory for <1 second. From when it's imported till when it's send over the websocket API.